### PR TITLE
WIP Make Text.new not require font

### PR DESCRIPTION
**Pull request by KillaPL**
So, what this PR attempts to do:
  - it updates Text class in such way, that it will work properly if user doesn't explicitly pass font

This is a good idea, because:
  - as a user, I want it to run without having to thinking about too many things
  - it makes it much easier to discuss about Text class in areas other then font, because `Font.new(0, 0, "text")` will work on any machine, where `Font.new(0, 0, "text", 20, "/path/to/font")` which is right now required as a minimal set of code to produce a text, will not, if you don't have font on that path.

What this PR does not:
  - It does not make sure that the font looks the same in every platform.
  - it does not validate correctness of passed in font path

  Just simply because those are other issues which we can solve separately. 
  I am more than happy to start discussion on how these goals can be achieved, I simply wanted to note that this has not been my primary goal for this pull request.

What works:
  - MRI on linux works well.

What doesn't work:
  - native on linux doesn't work (crashes). I am fairly certain that this is because `Dir#[]` is lacking in mruby. Here I probably need help.
  - web doesn't work too well (looks bad). I am fairly certain that https://github.com/simple2d/simple2d.js/pull/2 will fix it.

So, to sum this up, this is my initial attempt to make dealing with Font class more pleasant. Please let me know what you think.

### CODE REVIEW

**Comment by blacktm** [comment]
Just rebased with ruby2d/master. Looks like `#text=, maps Number to string` test is segfaulting, not sure why yet.

**Comment by KillaPL** [comment]
Updated with all recently merged code, compiles fine and all the tests pass nicely locally. Still has mentioned problems in native and web tho.

**Comment by blacktm** [comment]
Hey, I was taking a look at this again, but looks like you deleted the branch from your forked repo. Do you want to try restoring it so we can continue working on it? I can also try pulling the change set into a new branch on this repo if not.

**Comment by KillaPL** [comment]
My bad, I occasionally swipe entire fork and create it from scratch, when git state becomes weird, or when I brake something. I created new repo with the entire set of changes here: https://github.com/ruby2d/ruby2d/pull/53

Additionally it includes the whitespace changes, sorry about that...

**Comment by KillaPL** [comment]
Closing this PR in favor of https://github.com/ruby2d/ruby2d/pull/53
